azurerm_local_network_gateway fails to create when bgp settings are supplied

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.4.5

### AzureRM Provider Version

3.53.0

### Affected Resource(s)/Data Source(s)

azurerm_local_network_gateway 

### Terraform Configuration Files

```hcl
resource "azurerm_local_network_gateway" "local-gw-primary" {
  name                = "vnet-lng-vxprimary"
  resource_group_name = data.azurerm_resource_group.network-rg.name
  location            = data.azurerm_resource_group.network-rg.location
  gateway_address     = local.lng_ip_pri
  address_space       = [""]
  bgp_settings {
     asn                 = local.lng_bgp_asn_pri
     bgp_peering_address = local.lng_bgp_peering_address_pri
  }
  
  lifecycle {
    ignore_changes     = [address_space]
  }
}
```


### Debug Output/Panic Output

```shell
azurerm_local_network_gateway.local-gw-primary: Creating...
╷
│ Error: Plugin did not respond
│
│   with azurerm_local_network_gateway.local-gw-primary,
│   on main.tf line 136, in resource "azurerm_local_network_gateway" "local-gw-primary":
│  136: resource "azurerm_local_network_gateway" "local-gw-primary" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-azurerm_v3.53.0_x5 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 150 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/network.expandLocalNetworkGatewayAddressSpaces(0xc00144d7a0?)
        github.com/hashicorp/terraform-provider-azurerm/internal/services/network/local_network_gateway_resource.go:271 +0x1d4
github.com/hashicorp/terraform-provider-azurerm/internal/services/network.resourceLocalNetworkGatewayCreateUpdate(0xc00222ba80, {0x6167c60?, 0xc0019dec00})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/network/local_network_gateway_resource.go:160 +0xb25
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x75cb7e0?, {0x75cb7e0?, 0xc00293eb70?}, 0xd?, {0x6167c60?, 0xc0019dec00?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:695 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0013ea380, {0x75cb7e0, 0xc00293eb70}, 0xc002049e10, 0xc00222b900, {0x6167c60, 0xc0019dec00})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0006e5890, {0x75cb7e0?, 0xc00293ea50?}, 0xc0017c5720)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc001a095e0, {0x75cb7e0?, 0xc00293e1b0?}, 0xc002171650)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x68ed9a0?, 0xc001a095e0}, {0x75cb7e0, 0xc00293e1b0}, 0xc0021715e0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000014000, {0x75dadc0, 0xc0010be1a0}, 0xc002058360, 0xc001262270, 0xbaafac0, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc000014000, {0x75dadc0, 0xc0010be1a0}, 0xc002058360, 0x0)
        google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-azurerm_v3.53.0_x5 plugin crashed!
```


### Expected Behaviour

Resource is created with BGP Settings applied

### Actual Behaviour

Receive a message that the provider has crashed. I can see in the crash log that it appears to be looking for the resource, as if it should already exist.

If I apply the resource without BPG settings, it applies just fine. And if I add the BGP settings and apply the changes, it works just fine. It just does not apply the first time with the BGP settings without crashing the provider.

### Steps to Reproduce

terraform init
terraform apply

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/network/local_network_gateway_resource.go
+++ b/internal/services/network/local_network_gateway_resource.go
@@ -11,6 +11,7 @@ import (
 	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/parse"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
 	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
 	"github.com/hashicorp/terraform-provider-azurerm/utils"
 	"github.com/tombuildsstuff/kermit/sdk/network/2022-07-01/network"
@@ -61,7 +62,8 @@ func resourceLocalNetworkGateway() *pluginsdk.Resource {
 				Type:     pluginsdk.TypeList,
 				Optional: true,
 				Elem: &pluginsdk.Schema{
-					Type: pluginsdk.TypeString,
+					Type:         pluginsdk.TypeString,
+					ValidateFunc: validation.StringIsNotEmpty,
 				},
 			},
 

